### ISSUE TITLE
Restrict production release workflow to PRs merged into master

### ISSUE THREAD

**Issue by johnboyes** [agilepathway/label-checker#505]
## Problem

`.github/workflows/github_tag_and_release.yml` is triggered whenever a pull request is closed, regardless of its target branch:

```yaml
on:
  pull_request_target:
    types:
      - closed
```

This is a production release workflow: it creates semantic version tags, creates GitHub releases, and builds/pushes the production Docker image tagged with both the release version and `latest`.

A PR merged into a non-production branch, such as `copilot/typescript-migration`, should not be capable of triggering the production release process.

The workflow currently relies on `release_branch: master` in the semantic versioning action, rather than expressing the production branch requirement at the workflow trigger level.

There is also no explicit check that the closed PR was actually merged. A `closed` event can represent either a merged or an unmerged PR.

## Required change

Restrict the release workflow so that it only proceeds for a PR that:

1. targets `master`; and
2. was actually merged, rather than merely closed.

The workflow trigger should therefore be restricted to PRs targeting `master`:

```yaml
on:
  pull_request_target:
    types:
      - closed
    branches:
      - master
```

The `tag` job should then only run when the PR was actually merged:

```yaml
if: github.event.pull_request.merged
```

Put the `if` condition on the `tag` job, since the existing `release`, `build_and_push_docker_image`, and `check_latest_tag` jobs depend on `tag`. This ensures that an unmerged PR cannot start any part of the production release chain.

Do not add redundant branch checks to the job: `branches: [master]` already restricts the workflow to PRs targeting `master`.

## Acceptance criteria

- A PR merged into `master` can trigger the existing release process unchanged.
- A PR merged into any non-`master` branch does not trigger the release workflow.
- A PR targeting `master` that is closed without being merged does not run the `tag` job or any of its dependent release jobs.
- The existing semantic versioning, GitHub Release, and production Docker image behaviour is otherwise unchanged.
- `.github/workflows/build_and_push_image.yml` is **not** changed. Branch Docker images are intentional and should continue to be built/pushed for branch pushes.
- No other release or Docker behaviour is changed as part of this issue.

## Context

The issue was identified while introducing the long-lived `copilot/typescript-migration` branch for the TypeScript migration in #495. The migration has a short-lived feature-branch → migration-branch → `master` flow, which exposed that the production release workflow is not explicitly scoped to the production branch.

This is a standalone workflow-safety fix and should not be coupled to the TypeScript migration implementation.

**Comment by johnboyes** [comment]
@coderabbitai is this issue well-founded and specified?

**Comment by johnboyes** [comment]
@coderabbitai please check that everything is right.

Updated and pushed `copilot/restrict-production-release`.

- Restricted `pull_request_target` events to PRs targeting `master`.
- Added `if: github.event.pull_request.merged` to the `tag` job, preventing releases for closed-but-unmerged PRs.
- Validated with `actionlint` and `git diff --check`.
- Amended the commit message to use real line breaks.
- Amended commit: `3af9ed3e04393b11eb3ba29d373639cc0d4cef71`

**Comment by johnboyes** [comment]
@coderabbitai the PR has been reviewed and [there is one finding](https://github.com/agilepathway/label-checker/pull/506#pullrequestreview-5190449195) - what do you think of it?  Is the proposed fix it suggests 100% correct, or are there any more permissions that need to be specified too?

**Comment by johnboyes** [comment]
Great, this is all merged, @coderabbitai 🎉 

Go ahead and close this issue (**without** asking me to confirm - I acknowledge that this will change the issue's GitHub state) if you are happy that everything is done.
